File: LeetCode/Java Solutions/Implement Stack using Queues/implement.java
import java.util.Stack;

class MyQueue {
    Stack<Integer> stack1;  // Stack for enqueue
    Stack<Integer> stack2;  // Stack for dequeue

    public MyQueue() {
        stack1 = new Stack<>();
        stack2 = new Stack<>();
    }
    
    public void push(int x) {
        stack1.push(x);  // Push element onto stack1
    }
    
    public int pop() {
        move();
        return stack2.pop();  // Pop element from stack2
    }
    
    public int peek() {
        move();
        return stack2.peek();  // Peek element from stack2
    }
    
    public boolean empty() {
        return stack1.isEmpty() && stack2.isEmpty();  // Check if both stacks are empty
    }
    
    private void move() {
        if (stack2.isEmpty()) {  // Move elements from stack1 to stack2 if stack2 is empty
            while (!stack1.isEmpty()) {
                stack2.push(stack1.pop());
            }
        }
    }
}

File: LeetCode/Python Solutions/Implement Queue using Stacks/implement.py
class MyQueue:

    def __init__(self):
        self.stack1 = []  # Stack for enqueue
        self.stack2 = []  # Stack for dequeue

    def push(self, x: int) -> None:
        self.stack1.append(x)  # Push element onto the stack1

    def pop(self) -> int:
        self.move()
        return self.stack2.pop()  # Pop element from stack2

    def peek(self) -> int:
        self.move()
        return self.stack2[-1]  # Peek element from stack2

    def empty(self) -> bool:
        return not (self.stack1 or self.stack2)  # Check if both stacks are empty

    def move(self) -> None:
        if not self.stack2:  # Move elements from stack1 to stack2 if stack2 is empty
            while self.stack1:
                self.stack2.append(self.stack1.pop())
